knimin/static/js/platemapper.js

```diff
@@ -0,0 +1,506 @@
+  $.get('/pm_sample_plate?plate_id=' + plate_id, function (data) {
+    obj.initialize(data);
+  })
+    .fail(function (jqXHR, textStatus, errorThrown) {
+      if(jqXHR.status === 404) {
+        var data = $.parseJSON(jqXHR.responseText)
+        $('<h3 class="warning">').html(data.message).appendTo(obj.target);
+      } else {
+        $('<div>').html(jqXHR.responseText).appendTo(obj.target);
+      }
+    });
+};
+
+/**
+ *
+ * Initializes the object after the GET query is completed
+ *
+ * @param {Object} data The data returned from the GET query
+ *
+ **/
+PlateMap.prototype.initialize = function (data) {
+    var study, color;
+
+    this.name = data.name;
+    this.createdOn = data.created_on;
+    this.createdBy = data.email;
+    this.notes = data.notes;
+    this.plateType = data.plate_type.notes;
+    this.rows = data.plate_type.rows;
+    this.cols = data.plate_type.cols;
+    this.studies = data.studies;
+
+    // Construct a dictionary keyed by sample, for easy access to the sample
+    // information
+    this.samples = {}
+    // This is a special list needed for initializing the sample autocompletion
+    this.autoCompleteSamples = []
+    // Iterate over all the studies
+    for (var idx = 0; idx < this.studies.length; idx++) {
+      study = this.studies[idx];
+      color = PlateMap._qiimeColors[idx];
+      // Iterate over all samples
+      for (var sample of study.samples.all) {
+        this.samples[sample] = {color: color, plates: []};
+        this.autoCompleteSamples.push({label: sample, category: study.title, color: color});
+      }
+      // Iterate over all plates to get the already plated samples
+      for (var plate in study.samples.plated) {
+        if (study.samples.plated.hasOwnProperty(plate)) {
+          this.samples[sample].plates.push(plate);
+        }
+      }
+    }
+
+    // Create a 2D array to store the per well information
+    this.wellInformation = new Array(this.rows);
+    for (var i = 0; i < this.wellInformation.length; i++) {
+      this.wellInformation[i] = new Array(this.cols);
+      for (var j = 0; j < this.wellInformation[i].length; j++) {
+        this.wellInformation[i][j] = {inputTag: undefined, comment: undefined};
+      }
+    }
+
+    this.drawPlate();
+};
+
+/**
+ *
+ * Updates the contents of the text area with a summary of the well comments
+ *
+ **/
+PlateMap.prototype.updateWellCommentsArea = function () {
+  var sample, wellId, wellInfo;
+  var comments = "";
+  for (var i = 0; i < this.rows; i++) {
+    for (var j = 0; j < this.cols; j++) {
+      wellInfo = this.wellInformation[i][j];
+      if (wellInfo.comment !== undefined) {
+        sample = wellInfo.inputTag.val().trim();
+        if (sample.length === 0) {
+          sample = 'No sample plated';
+        }
+        wellId = String.fromCharCode('A'.charCodeAt() + i) + (j + 1);
+        comments = comments + "Well " + wellId + " (Sample: " + sample + "): " + wellInfo.comment + "\n";
```

Comment:
Is the \n accurate, I think this needs a <br> instead? Also, I would recommend instead a list of comments, vs a long string of multi-line comments, but that's minor.